tmail-backend/rate-limiter/rate-limiter-cassandra/src/main/scala/com/linagora/tmail/rate/limiter/cassandra/CassandraRateLimitPlanDAO.scala

```diff
@@ -0,0 +1,125 @@
+package com.linagora.tmail.rate.limiter.cassandra
+
+import com.datastax.driver.core.querybuilder.QueryBuilder
+import com.datastax.driver.core.querybuilder.QueryBuilder.{bindMarker, insertInto, select}
+import com.datastax.driver.core.{PreparedStatement, Row, Session, UDTValue}
+import com.linagora.tmail.rate.limiter.api.{OperationLimitations, RateLimitingPlanCreateRequest, RateLimitingPlanId, RateLimitingPlanResetRequest}
+import com.linagora.tmail.rate.limiter.cassandra.CassandraRateLimitPlanDAO.{LimitingEntry, RateLimitingPlanEntry}
+import com.linagora.tmail.rate.limiter.cassandra.CassandraRateLimitPlanTable.{OPERATION_LIMITATION_NAME, PLAN_ID, PLAN_NAME, RATE_LIMITATION_BASE, RATE_LIMITATION_MAPPING, RateLimitationBase, TABLE_NAME}
+import org.apache.james.backends.cassandra.init.CassandraTypesProvider
+import org.apache.james.backends.cassandra.utils.CassandraAsyncExecutor
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.jdk.CollectionConverters._
+
+object CassandraRateLimitPlanDAO {
+  object RateLimitingPlanEntry {
```

Comment:
I have doubts regarding nested object.

Being able to do it do not mean that we should...

Reply:
I just want to notice the purpose of `RateLimitingPlanEntry` is to support `CassandraRateLimitPlanDAO`